Fixed reasons

cjduffett opened the pull request:
Swapped the code-display properties that were incorrect in the injuries module. Removed the reason attribute in the Procedure states in the allergic rhinitis module since we can't conclusively point that reason to a single ConditionOnset state. It would probably be a meaningful enhancement to allow reasons to be attributes as well.

dehall commented:
The reason being set to an attribute should be fine - this is how it's handled in the code: https://github.com/synthetichealth/synthea/blob/master/lib/generic/states.rb#L414  The real issue described in #96 is that we're inconsistent with how "reason" is used across states, and the wiki isn't clear on that.

cjduffett commented:
I added a `reason` to every `Encounter`, `Procedure`, `MedicationOrder`, and `CarePlanStart` state where a reason could possibly apply. I also changed the ordering of fields in a few of the GMF states for consistency.

Transitions are now the last field in every GMF state. All remarks are now lists, for consistency. I'm going to run a JSON formatter in a separate commit to clean things up further.

cjduffett commented:
Ready to merge
